src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -527,7 +543,7 @@ internal void WriteCentralDirectoryFileHeader()
             writer.Write(_fileComment != null ? (ushort)_fileComment.Length : (ushort)0); // file comment length
             writer.Write((ushort)0); // disk number start
             writer.Write((ushort)0); // internal file attributes
-            writer.Write((uint)0); // external file attributes
+            writer.Write(_externalFileAttr); // external file attributes
```

Comment:
It's not, really. Testing that the zips written work correctly with other programs requires manual testing (which FWIW I did).

As @eerhardt mentioned below, I can add a test to do a more complete roundtripping test to hit this line.